prometheus/src/main/java/org/opensearch/sql/prometheus/client/PrometheusClientImpl.java

```diff
@@ -111,14 +113,21 @@ private List<String> toListOfLabels(JSONArray array) {
 
   private JSONObject readResponse(Response response) throws IOException {
     if (response.isSuccessful()) {
-      JSONObject jsonObject = new JSONObject(Objects.requireNonNull(response.body()).string());
+      JSONObject jsonObject;
+      try {
+        jsonObject = new JSONObject(Objects.requireNonNull(response.body()).string());
+      } catch (JSONException jsonException) {
+        throw new PrometheusClientException(
+            "Prometheus returned unexpected body, "
+                + "please verify your prometheus server setup.");
+      }
       if ("success".equals(jsonObject.getString("status"))) {
         return jsonObject;
       } else {
-        throw new RuntimeException(jsonObject.getString("error"));
+        throw new PrometheusClientException(jsonObject.getString("error"));
```

Comment:
Do you have a test for this case?

Reply:
https://github.com/opensearch-project/sql/blob/main/prometheus/src/test/java/org/opensearch/sql/prometheus/client/PrometheusClientImplTest.java#L70C1-L70C1